src/main/java/org/testng/internal/Invoker.java

```diff
@@ -430,8 +430,24 @@ private boolean confInvocationPassed(ITestNGMethod method, ITestNGMethod current
         if (! m_continueOnFailedConfiguration) {
           result = !classConfigurationFailed(cls);
         } else {
-          synchronized(m_classInvocationResults){
-             result = !m_classInvocationResults.get(cls).contains(instance);
+          synchronized (m_classInvocationResults) {
+            Set<Object> set = m_classInvocationResults.get(cls);
+            //If the set was null, it means that the test method is residing in a child class
+            //and maybe the parent method has configuration methods which may have had a failure
+            //So lets walk up the inheritance tree until either we find failures or till we
+            //reached the Object class.
+            while (set == null || cls.equals(Object.class)) {
```

Comment:
`while (set == null || !cls.equals(Object.class))`, true?

Reply:
My bad!! Fixed, amended commit and re-pushed. Please check